src/org/zaproxy/zap/extension/ascanrulesAlpha/TestNoSQLInjection.java

```diff
@@ -0,0 +1,651 @@
+								sendAndReceive(injectedMsg, false);
+								intervalInjectedMessage = ChronoUnit.MILLIS.between(now, Instant.now());
+																
+								if(isTimedInjected(intervalBaseMessage,intervalInjectedMessage, SLEEP_TIME_SHORT)) {
+									
+									// try for a longer time to exclude transmission delays  
+									
+									injectedMsg = getNewMsg();
+									valueInjected =tvi[1];
+									setParameter(injectedMsg, param, valueInjected);
+									
+									now = Instant.now();
+									sendAndReceive(injectedMsg, false);
+									intervalInjectedMessage = ChronoUnit.MILLIS.between(now, Instant.now());
+									
+									if(isTimedInjected(intervalBaseMessage,intervalInjectedMessage, SLEEP_TIME_LONG)) {
+										
+										String extraInfo = Constant.messages.getString(EXTRAINFO_PREFIX + nosql.prefix 
+												+ attackType);
+										String name = getName() + " - " + nosql.getName();
+										
+										bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_HIGH, name, getDescription(), null, 
+												param, valueInjected, extraInfo,  getSolution(), injectedMsg);
+										break;
+									}
+								}
+								
+							} catch (SocketException ex) {
+		
+								log.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 
+										" when accessing: " + msg.getRequestHeader().getURI().toString());
+								continue;
+							}
+							finally {
+								if(isStop())
+									return;
+							}						
+						}
+
+					}
+					
+					/**
+					 * TODO is content-type json format?
+					 * 
+					 */
+					boolean jsonEnable = false;
+					if (jsonEnable) {
+						
+						for(String[] jpv : nosql.getJsonParamValueInjection()) {
+							
+							onlyExactMatch = false;
+							attackType = JSON_ATTACK;
+							injectedMsg = getNewMsg();
+							
+							
+							/**
+							 * TODO  set post request with json content-type
+							 * 
+							 */
+							injectedMsg.getRequestHeader().setHeader(HttpRequestHeader.CONTENT_TYPE, 
+									"application/json");
+							injectedMsg.getRequestHeader().setMethod(HttpRequestHeader.POST);
+							
+							try {
+								valueInjected = getParamJsonString(param, jpv);
+								setParameter(injectedMsg, param, valueInjected);
+								sendAndReceive(injectedMsg, false);
+								if(isBingo(baseMsg, injectedMsg, param, valueInjected, attackType, nosql, 
+										onlyExactMatch)) {
+									break;
+								}
+							}catch(SocketException e) {
+								log.debug("Caught " + e.getClass().getName() + " " + e.getMessage() + 
+								" when accessing: " + msg.getRequestHeader().getURI().toString());
+								continue;
+							}catch(JSONException e) {
+								log.debug("Caught " + e.getClass().getName() + " " + e.getMessage() + 
+								" when try to convert to json the value: " + jpv);
+								continue;
+							}
+							finally {
+								if(isStop())
+									return;
+							}
+						}
+					}
+	    		}
+			}
+		} catch (IOException e) {
+			e.printStackTrace();

```

Comment:
Are there more specific exceptions that could/should be caught?
This should use the logger instead of just printing.